### ISSUE TITLE
No errors found in deformed html-doc

### ISSUE THREAD

**Issue by toben101** [validator/validator#423]
Hi there,
I'm not an expert in html, but when checking the following doc from one of my pupils today, I got no errors reported. That seems strange to me.
Thanks in advance.
t

<!DOCTYPE html>
<html lang="de">
<head>
<title>Einführung in HTML5 (und CSS)</title>
</body>
</html>

**Comment by sideshowbarker** [comment]
Unless the document also starts with `<!doctype html>` the checker will report an error.

But the with the <!doctype html>` included it has no errors:

```html
<!DOCTYPE html>
<title>Einführung in HTML5 (und CSS)</title>
```

That’s because for the `html`, `head`, and `body` elements, the star tags and end tags can be omitted.

The only required element every HTML document must have is a non-empty `title` element.

**Comment by toben101** [comment]
Hi sideshowbarker,
when reading my post now, there is information missing (is it monday today?). Sorry!
The Document I wanted to discuss is this (no errors in validator):

<!DOCTYPE html>
<html lang="de">
<head>
<title>Einführung in HTML5 (und CSS)</title>
</body>
</html>

Thanks in advance.

**Comment by sideshowbarker** [comment]
OK thanks. So what error would you expect the checker to report for that document?

**Comment by toben101** [comment]
Perhaps:

head-tag opened, but not closed.
body-tag closed, but not opened.

**Comment by sideshowbarker** [comment]
> head-tag opened, but not closed.
> body-tag closed, but not opened.

The [end tag for the `head` element is optional](https://html.spec.whatwg.org/multipage/semantics.html#the-head-element); if omitted, that’s not an error (same for its start tag).

The [start tag for the `body` element is optional](https://html.spec.whatwg.org/multipage/semantics.html#the-body-element); if omitted, that’s not an error (same for its end tag).

**Comment by cvrebert** [comment]
Tools such as HTMLHint or htmllint can be used to detect missing explicit opening/closing tags.

**Comment by sideshowbarker** [comment]
Closing per https://github.com/validator/validator/issues/423#issuecomment-265456670
